src/dataDisplay/table.jsx

```diff
@@ -1,21 +1,350 @@
+        tableStickyColumns: PropTypes.shape({}),
+    }),
+    className: PropTypes.string,
+    collapsing: PropTypes.bool,
+    definition: PropTypes.bool,
+    fixed: PropTypes.bool,
+    fontSize: PropTypes.oneOf(Utils.sizeEnums()),
+    fullWidth: PropTypes.bool,
+    id: PropTypes.string,
+    selectable: PropTypes.bool,
+    singleLine: PropTypes.bool,
+    size: PropTypes.oneOf(['l', 'large', 'm', 'medium', 's', 'small']),
+    stackable: PropTypes.bool,
+    stickyColumnCount: PropTypes.number,
+    resizableColumnWidthPercentage: (props, propName, componentName) => {
+        const {
+            [propName]: propValue,
+        } = props;
+        const isValid = propValue > 0 &&
+            propValue <= parseInt(MAX_WIDTH_PERCENTAGE_DESKTOP * 100, 10);
+        if (!isValid) {
+            return new Error(
+                `Invalid prop value for ${propName} (${propValue}) supplied to ${componentName}. Validation failed.`,
+            );
+        }
+        return null;
+    },
+    stretch: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.oneOf(['very']),
+    ]),
+    striped: PropTypes.bool,
+    style: PropTypes.shape({}),
+};
+
+const defaultProps = {
+    basic: undefined,
+    celled: undefined,
+    children: undefined,
+    classes: undefined,
+    className: undefined,
+    collapsing: undefined,
+    definition: undefined,
+    fixed: undefined,
+    fontSize: undefined,
+    fullWidth: undefined,
+    id: undefined,
+    selectable: undefined,
+    singleLine: undefined,
+    size: undefined,
+    stackable: undefined,
+    stickyColumnCount: 0,
+    resizableColumnWidthPercentage: undefined,
+    stretch: undefined,
+    striped: undefined,
+    style: undefined,
+};
+
+const DEBOUNCE_WAIT_TIME = 50;
+const TABLE_HEADER_TYPE = 'TableHeader';
+const STICKY_CELL_CLASS = 'sticky-cell';
+const STICKY_CELL_RESIZABLE_CLASS = 'sticky-cell-resizable';
+const STICKY_CELL_FIRST_OF_ROW_CLASS = 'sticky-cell-first-of-row';
+const STICKY_CELL_LAST_OF_COLUMN_CLASS = 'sticky-cell-last-of-column';
+const STICKY_CELL_BORDER_STYLE = `1px solid ${borderColor.default}`;
+
+class Table extends React.PureComponent {
+    constructor() {
+        super();
+        this.state = {
+            adjacentStickyColumnsTotalWidth: 0,
+            stickyTableContainerWidth: 0,
+        };
+        this.applyStickyClassesToCells = this.applyStickyClassesToCells.bind(this);
+        this.delayedSetScrollBarPosition = null;
+        this.forceTableUpdate = this.forceTableUpdate.bind(this);
+        this.parseChildren = this.parseChildren.bind(this);
+        this.setResizableCellsWordWrapping = this.setResizableCellsWordWrapping.bind(this);
+        this.setStickyCellsStylesOnScroll = this.setStickyCellsStylesOnScroll.bind(this);
+    }
+
+    componentDidMount() {
+        this.setStickyColumnPositions();
+        this.applyStickyClassesToCells();
+        this.delayedSetStickyTableContainerWidth = debounce(() => {
```

Comment:
Is this debounce used if `stickyColumnCount` is > 0?

Reply:
No, it's not, I'm applying a validation so we create the debounce only if the `stickyColumnCount > 0`